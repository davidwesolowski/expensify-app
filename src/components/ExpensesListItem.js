import React from 'react';
import { Link } from 'react-router-dom';

const ExpensesListItem = (props) => (
    <div>
        <h3><Link to={`edit/${props.expense.id}`}>{props.expense.description}</Link></h3>
        <p>
            {props.expense.amount} - {props.expense.createdAt}
        </p>    
        
    </div>
);

export default ExpensesListItem;