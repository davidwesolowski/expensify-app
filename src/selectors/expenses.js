import moment from 'moment';

const getVisibleExpenses = (expenses, { text, sortedBy, startDate, endDate }) =>
{
    return expenses.filter((expense) =>
    {
        const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());
        // const startDateMatch = startDate ? expense.createdAt >= startDate :true;
        // const endDateMatch = endDate ? expense.createdAt <= endDate : true;
        const createdAtMoment = moment(expense.createdAt);
        const startDateMatch = startDate ? moment(startDate).isSameOrBefore(createdAtMoment) : true;
        const endDateMatch = endDate ? moment(endDate).isSameOrAfter(createdAtMoment) : true;
        return textMatch && startDateMatch && endDateMatch;
    }).sort((a, b) =>
    {
        if (sortedBy === 'date')
        {
            return a.createdAt < b.createdAt ? 1 : -1;
        }
        else if (sortedBy === 'amount')
        {
            return a.amount < b.amount ? 1 : -1;
        }
    });
}

export { getVisibleExpenses };