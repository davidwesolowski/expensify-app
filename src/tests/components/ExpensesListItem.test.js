import React from 'react';
import { shallow } from 'enzyme';
import ExpensesListItem from '../../components/ExpensesListItem';
import expenses from '../fixtures/expenses';

test('Should render ExpensesListItem', () =>
{
    const wrapper = shallow(<ExpensesListItem expense={expenses[0]}/>)
    expect(wrapper).toMatchSnapshot();
});