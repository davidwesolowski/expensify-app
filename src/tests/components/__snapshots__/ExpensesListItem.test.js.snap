// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesListItem 1`] = `
<div>
  <h3>
    <Link
      to="edit/1"
    >
      Gum
    </Link>
  </h3>
  <p>
    195
     - 
    0
  </p>
</div>
`;
